From: Jon Masters <jcm@redhat.com>
Subject: [RHEL5.1 PATCH] BZ252377 - Need to verify list of EXPORT_SYMBOLS
Date: Mon, 27 Aug 2007 14:19:19 -0400
Bugzilla: 252377
Message-Id: <1188238759.1322.343.camel@jcmlaptop>
Changelog: [misc] re-export some symbols as EXPORT_SYMBOL_GPL


Specifically, the following 6 symbols are changed to GPL exports:

+EXPORT_SYMBOL_GPL(scsi_eh_get_sense);
+EXPORT_SYMBOL_GPL(scsi_eh_ready_devs);
+EXPORT_SYMBOL_GPL(kmap_flush_unused);
+EXPORT_SYMBOL_GPL(zap_page_range);
+EXPORT_SYMBOL_GPL(skb_checksum_setup);
+EXPORT_SYMBOL_GPL(sysctl_xfrm_acq_expires);

Jon.


diff -urNp linux-2.6.18.noarch/drivers/scsi/scsi_error.c linux-2.6.18.symexports/drivers/scsi/scsi_error.c
--- linux-2.6.18.noarch/drivers/scsi/scsi_error.c	2007-08-27 13:18:59.000000000 -0400
+++ linux-2.6.18.symexports/drivers/scsi/scsi_error.c	2007-08-27 13:36:25.000000000 -0400
@@ -831,7 +831,7 @@ int scsi_eh_get_sense(struct list_head *
 
 	return list_empty(work_q);
 }
-EXPORT_SYMBOL(scsi_eh_get_sense);
+EXPORT_SYMBOL_GPL(scsi_eh_get_sense);
 
 /**
  * scsi_eh_tur - Send TUR to device.
@@ -1429,7 +1429,7 @@ void scsi_eh_ready_devs(struct Scsi_Host
 				if (!scsi_eh_host_reset(work_q, done_q))
 					scsi_eh_offline_sdevs(work_q, done_q);
 }
-EXPORT_SYMBOL(scsi_eh_ready_devs);
+EXPORT_SYMBOL_GPL(scsi_eh_ready_devs);
 
 /**
  * scsi_eh_flush_done_q - finish processed commands or retry them.
diff -urNp linux-2.6.18.noarch/mm/highmem.c linux-2.6.18.symexports/mm/highmem.c
--- linux-2.6.18.noarch/mm/highmem.c	2007-08-27 13:18:26.000000000 -0400
+++ linux-2.6.18.symexports/mm/highmem.c	2007-08-27 13:32:17.000000000 -0400
@@ -150,7 +150,7 @@ void kmap_flush_unused(void)
 	spin_unlock(&kmap_lock);
 }
 
-EXPORT_SYMBOL(kmap_flush_unused);
+EXPORT_SYMBOL_GPL(kmap_flush_unused);
 #endif
 
 void fastcall *kmap_high(struct page *page)
diff -urNp linux-2.6.18.noarch/mm/memory.c linux-2.6.18.symexports/mm/memory.c
--- linux-2.6.18.noarch/mm/memory.c	2007-08-27 13:18:49.000000000 -0400
+++ linux-2.6.18.symexports/mm/memory.c	2007-08-27 13:34:36.000000000 -0400
@@ -893,7 +893,7 @@ unsigned long zap_page_range(struct vm_a
 		tlb_finish_mmu(tlb, address, end);
 	return end;
 }
-EXPORT_SYMBOL(zap_page_range);
+EXPORT_SYMBOL_GPL(zap_page_range);
 
 /*
  * Do a quick page-table lookup for a single page.
diff -urNp linux-2.6.18.noarch/net/core/dev.c linux-2.6.18.symexports/net/core/dev.c
--- linux-2.6.18.noarch/net/core/dev.c	2007-08-27 13:18:26.000000000 -0400
+++ linux-2.6.18.symexports/net/core/dev.c	2007-08-27 13:31:48.000000000 -0400
@@ -3646,7 +3646,7 @@ EXPORT_SYMBOL(unregister_netdevice_notif
 EXPORT_SYMBOL(net_enable_timestamp);
 EXPORT_SYMBOL(net_disable_timestamp);
 EXPORT_SYMBOL(dev_get_flags);
-EXPORT_SYMBOL(skb_checksum_setup);
+EXPORT_SYMBOL_GPL(skb_checksum_setup);
 
 #if defined(CONFIG_BRIDGE) || defined(CONFIG_BRIDGE_MODULE)
 EXPORT_SYMBOL(br_handle_frame_hook);
diff -urNp linux-2.6.18.noarch/net/xfrm/xfrm_state.c linux-2.6.18.symexports/net/xfrm/xfrm_state.c
--- linux-2.6.18.noarch/net/xfrm/xfrm_state.c	2007-08-27 13:18:51.000000000 -0400
+++ linux-2.6.18.symexports/net/xfrm/xfrm_state.c	2007-08-27 14:03:36.000000000 -0400
@@ -32,7 +32,7 @@ u32 sysctl_xfrm_aevent_rseqth __read_mos
 EXPORT_SYMBOL(sysctl_xfrm_aevent_rseqth);
 
 u32 sysctl_xfrm_acq_expires __read_mostly = 30;
-EXPORT_SYMBOL(sysctl_xfrm_acq_expires);
+EXPORT_SYMBOL_GPL(sysctl_xfrm_acq_expires);
 
 /* Each xfrm_state may be linked to two tables:
 

